src/request.rs

```diff
@@ -16,6 +16,14 @@ use std::sync::Arc;
 use std::time::Duration;
 
 const LOCK_RETRY_DELAY_MS: u64 = 10;
+const REGION_RETRY_DEFAULT_DELAY_MS: u64 = 500;
+const REGION_RETRY_MAX_COUNT: usize = 3;
+
+#[derive(Clone, Copy)]
+pub struct RetryState {
+    region_retry_count: usize,
+    region_retry_delay_ms: u64,
```

Comment:
It looks like the delay is never changed, so we should just be able to use `REGION_RETRY_MAX_COUNT` and remove this field